feat: make sli interval=lookback window for SLI code bundles

hrithvikakb opened the pull request.

hrithvikakb committed:
feat: make sli interval=lookback window for SLI code bundles

Rohit-Ekbote commented on codebundles/azure-aks-triage/sli.robot:
```diff
@@ -87,6 +88,16 @@ Generate AKS Cluster Health Score
     RW.Core.Push Metric    ${health_score}
 
 *** Keywords ***
+Normalize Lookback Window
```
Check if we can handle this in `RW.Core.Import Platform Variable`.

hrithvikakb committed:
chore: address PR comments

stewartshea committed:
enhance http-ok with verbosity and default to insecure checks (#582)

stewartshea committed:
Updates/251020-01 (#584)

* Update URL handling in HTTP and TLS templates to conditionally append paths based on presence of parentheses. This enhancement improves flexibility in URL configurations across multiple YAML templates, ensuring accurate path representation for both HTTP and HTTPS protocols.

* Enhance service bus connectivity test script by removing the port number from the hostname extraction. This change improves the clarity of the Service Bus endpoint hostname output.

* Refactor URL generation in HTTP and TLS YAML templates to improve path handling. Updated logic to conditionally extract inner paths from parentheses and ensure proper formatting for both HTTP and HTTPS protocols. This change enhances the clarity and maintainability of URL configurations across multiple templates.

* Refactor URL generation logic in HTTP and TLS YAML templates to improve path handling. Updated the logic to conditionally extract prefixes from paths, ensuring proper formatting for both HTTP and HTTPS protocols. This change enhances the clarity and maintainability of URL configurations across multiple templates.

stewartshea committed:
Updates/251020-02 (#585)

* Enhance Kubernetes deployment operations with new HPA and resource management tasks

- Introduced tasks for scaling up and down Horizontal Pod Autoscalers (HPA) for deployments, allowing for dynamic adjustment of min/max replicas based on traffic demands.
- Added functionality to intelligently increase and decrease CPU and memory resources for deployments, leveraging Vertical Pod Autoscaler (VPA) recommendations and ensuring optimal resource utilization.
- Updated README and runbook documentation to reflect new capabilities and configuration options, enhancing clarity for users managing Kubernetes deployments.
- Implemented safety checks to prevent conflicts with existing HPA configurations and GitOps-managed deployments, ensuring smooth resource management operations.

* Enhance HPA health checks and scaling tasks for Kubernetes deployments

- Added comprehensive health check tasks for Horizontal Pod Autoscalers (HPA), validating both configuration and runtime status, including checks for min/max replicas, resource requests, and scaling behavior.
- Updated runbook and README documentation to include new HPA health check details, improving clarity for users managing Kubernetes deployments.
- Implemented GitOps management checks in HPA scaling tasks to ensure changes are suggested rather than applied directly when managed by GitOps tools like Flux or ArgoCD.
- Enhanced error handling and reporting for HPA scaling operations, providing clearer guidance for users on potential issues and next steps.

* Update log pattern categories and documentation in healthcheck runbook and SLI scripts

- Modified the example and default log pattern categories in the runbook to remove 'StackTrace', streamlining the focus on critical error patterns.
- Updated documentation in the task description to reflect the changes in log analysis, ensuring clarity on the types of issues being monitored.
- Adjusted the critical error categories in the SLI script to align with the updated focus, enhancing the efficiency of SLI checks.

* Enhance resource scaling tasks in Kubernetes deployment operations

- Updated README to include detailed support for CPU and memory formats, ensuring proper handling of various Kubernetes resource specifications.
- Refactored runbook to improve CPU and memory resource calculations, incorporating support for millicores and different memory units (Mi, Gi, M, G).
- Enhanced logic for scaling up and down resources, ensuring accurate conversions and safety checks for resource limits.
- Improved documentation clarity regarding resource management tasks, aiding users in understanding scaling operations.

* Refactor HPA scaling tasks in Kubernetes runbook for improved clarity and functionality

- Streamlined the logic for scaling up and down Horizontal Pod Autoscalers (HPA), enhancing checks for GitOps management and ensuring accurate reporting of scaling operations.
- Improved error handling and reporting for HPA updates, providing clearer guidance on potential issues and next steps.
- Updated documentation within the runbook to reflect changes in HPA management tasks, aiding users in understanding scaling operations and configurations.
- Enhanced logging to capture commands used during HPA scaling, improving traceability and debugging capabilities.

* Refactor HPA scaling logic in Kubernetes runbook for improved flow and clarity

- Adjusted the conditional structure for updating Horizontal Pod Autoscalers (HPA) to enhance readability and maintainability.
- Ensured that command history is consistently logged after HPA updates, improving traceability of actions taken during scaling operations.
- Updated documentation within the runbook to reflect these changes, aiding users in understanding the updated scaling process.

stewartshea committed:
Patch/http-ok-hierarchy (#586)

* add hiearchy and tags to httpok

* fix access labels

hrithvikakb committed:
feat: update to emit observed at timestamp for issues (#581)

* feat: update to emit observed at timestamp for issues

* feat: address PR comments

stewartshea committed:
Update README.md

bump

hrithvikakb committed:
feat: use RW_LOOKBACK_WINDOW in scripts
